File: Python/Transactions/Update_Transaction.py
##
# BluePay Python Sample code.
#
# This code sample runs a $3.00 Credit Card Sale transaction
# against a customer using test payment information. If
# approved, a 2nd transaction is run to update the first transaction 
# to $5.75, $2.75 more than the original $3.00.
# If using TEST mode, odd dollar amounts will return
# an approval and even dollar amounts will return a decline.
##
from __future__ import print_function
import os.path, sys
sys.path.append(os.path.join(os.path.dirname(os.path.realpath(__file__)), os.pardir))
from BluePay import BluePay

account_id = "Merchant's Account ID Here"
secret_key = "Merchant's Secret Key Here"
mode = "TEST"  

payment = BluePay(
    account_id = account_id, 
    secret_key = secret_key, 
    mode = mode
)

payment.set_customer_information(
    name1 = "Bob",
    name2 = "Tester",
    addr1 = "123 Test St.",
    addr2 = "Apt #500",
    city = "Testville",
    state = "IL",
    zipcode = "54321",
    country = "USA"
)

payment.set_cc_information(
    card_number = "4111111111111111",
    card_expire = "1225",
    cvv2 = "123"
)

payment.sale(amount = '3.00') # Sale Amount: $3.00

# Makes the API request for processing the sale
payment.process()

# If transaction was approved..
if payment.is_successful_response():

    payment_update = BluePay(
        account_id = account_id, # Merchant's Account ID
        secret_key = secret_key, # Merchant's Secret Key
        mode = mode # Transaction Mode: TEST (can also be LIVE)
    )

    # Creates an update transaction against previous sale
    payment_update.update(
        transaction_id = payment.trans_id_response, # id of the transaction to update
        amount = '5.75' # add $2.75 to previous amount
    )

    # Make the API request
    payment_update.process()

    # Read response from BluePay
    print('Transaction Status: ' + payment_update.status_response)
    print('Transaction Message: ' + payment_update.message_response)
    print('Transaction ID: ' + payment_update.trans_id_response)
    print('AVS Response: ' + payment_update.avs_code_response)
    print('CVV2 Response: ' + payment_update.cvv2_code_response)
    print('Masked Payment Account: ' + payment_update.masked_account_response)
    print('Card Type: ' + payment_update.card_type_response)
    print('Auth Code: ' + payment_update.auth_code_response)
else:
    print(payment_update.message_response)

File: C++/Transactions/Update_Transaction.cpp
//
// BluePay C++ Sample code.
//
// This code sample runs a $3.00 Credit Card Sale transaction
// against a customer using test payment information. If
// approved, a 2nd transaction is run to update the first transaction
// to $5.75, $2.75 more than the original $3.00.
// If using TEST mode, odd dollar amounts will return
// an approval and even dollar amounts will return a decline.


#include "Update_Transaction.h"
#include "../bluepay-cpp/BluePay.h"
#include <iostream>
using namespace std;

void updateTransaction(){
	string accountId = "Merchant's Account ID Here";
	string secretKey = "Merchant's Secret Key Here";
	string mode = "TEST";

	BluePay payment(
	  	accountId, 
	    secretKey, 
	    mode 
	);

	 payment.setCustomerInformation(
	    "Bob", // First Name
	    "Tester", // Last Name
	    "123 Test St.", // Address1
	    "Apt #500", // Address2
	    "Testville", // City
	    "IL", // State
	    "54321", // Zip
	    "USA", // Country
	    "1231231234", // Phone Number
	    "test@bluepay.com" // Email Address
	 );

	 payment.setCCInformation(
	 	"4111111111111111", // Card Number
	    "1225", // Card Expire
	    "123" // Card CVV2
	 );

	// Sale Amount: $3.00
	payment.sale("3.00");

	payment.process();


	// Reads the responses from BluePay if transaction was approved
	if (payment.isSuccessfulTransaction()){
		
		BluePay paymentUpdate(
			accountId,
			secretKey,
			mode
		);

		paymentUpdate.update(payment.getTransId(), "5.75"); // add $2.75 to previous amount

		paymentUpdate.process();

		// Reads the responses from BluePay 
		cout << string("Transaction Status: ") + paymentUpdate.getResult() + "\n";
		cout << string("Transaction Message: ") + paymentUpdate.getMessage() + "\n";
		cout << string("Transaction ID: ") + paymentUpdate.getTransId() + "\n";
		cout << string("AVS Result: ") + paymentUpdate.getAvs() + "\n";
		cout << string("CVV2 Result: ") + paymentUpdate.getCvv2() + "\n";
		cout << string("Masked Payment Account: ") + paymentUpdate.getMaskedPaymentAccount() + "\n";
		cout << string("Card Type: ") + paymentUpdate.getCardType() + "\n";
		cout << string("Authorization Code: ") + paymentUpdate.getAuthCode() + "\n";
	}
	else {
        cout << string("Error: ") + payment.getMessage();
	}
}
